snorkel/learning_utils.py

```diff
@@ -193,18 +193,23 @@ class GridSearch(object):
     Runs hyperparameter grid search over a Learner object with train and test methods
     Selects based on maximizing F1 score on a supplied cross-validation (cv) set.
     """
-    def __init__(self, learner, param_names, param_val_ranges):
-        self.learner           = learner
-        self.param_names       = param_names
-        self.param_val_ranges  = param_val_ranges
-
-    def fit(self, candidates, gold_labels, **model_hyperparams):
+    def __init__(self, model, X, training_marginals, param_names, param_val_ranges):
+        self.model              = model
+        self.X                  = X
+        self.training_marginals = training_marginals
+        self.param_names        = param_names
+        self.param_val_ranges   = param_val_ranges
+        
+    def search_space(self):
+        return product(*self.param_val_ranges)
+
+    def fit(self, X_test, L_test, gold_candidate_set, b=0.5, set_unlabeled_as_neg=True, **model_hyperparams):
```

Comment:
Some minor things:
* I would call e.g. `X_validation` to make it clear that they shouldn't use the test set here?
* `L_test` is a vector of labels, so should be e.g. `validation_labels` (upper case single letter reserved for matrices)